src/argilla/labeling/text_classification/rule.py

```diff
@@ -74,6 +74,40 @@ def author(self):
         """Who authored the rule."""
         return self._author
 
+    def convert_to_labeling_rule(self):
+        """Converts the rule to a LabelingRule"""
+        if isinstance(self._label, str):
+            labels = [self._label]
+        else:
+            labels = self._label
+
+        return LabelingRule(
+            query=self.query, labels=labels, author=self.author or "None"
+        )
+
+    def update_label(self, label):
+        """Updates the label"""
+        self._label = label
+
+    def add_to_dataset(self, dataset: str):
+        """Add to rule to the given dataset"""
+        api.active_api().add_dataset_labeling_rules(
+            dataset, rules=[self.convert_to_labeling_rule()]
+        )
+
+    def remove_from_dataset(self, dataset: str):
+        """Removes the rule from the given dataset"""
+
+        api.active_api().delete_dataset_labeling_rules(
+            dataset, rules=[self.convert_to_labeling_rule()]
+        )
+
+    def update_at_dataset(self, dataset: str):
+        """Updates the rule at the given dataset"""
+        api.active_api().update_dataset_labeling_rules(
+            dataset, rules=[self.convert_to_labeling_rule()]
+        )
```

Comment:
So, this method will allow also rules individually, isn't it?

```python
rule = Rule(...)

rule.add_to_dataset(ds)

rule.label = ...
rule.update_at_dataset(ds)

````

For me, it is a bit weird here without a dataset container class.

We can create a rule and update it to a dataset where does not exists.

```python
rule = Rule(...)
rule.update_at_dataset(ds)
```

But not sure if could be helpful in the flow.  @dvsrepo what do you think?